libbeat/common/docker/client.go

```diff
@@ -28,31 +29,35 @@ import (
 	"golang.org/x/net/context"
 )
 
-// Select Docker API version
-const dockerAPIVersion = api.DefaultVersion
-
 // NewClient builds and returns a new Docker client
-// It uses version 1.26 by default, and negotiates it with the server so it is downgraded if 1.26 is too high
+// It uses version 1.30 by default, and negotiates it with the server so it is downgraded if 1.30 is too high
 func NewClient(host string, httpClient *http.Client, httpHeaders map[string]string) (*client.Client, error) {
-	c, err := client.NewClient(host, dockerAPIVersion, httpClient, nil)
-	if err != nil {
-		return c, err
+	version := os.Getenv("DOCKER_API_VERSION")
+	if version == "" {
+		version = api.DefaultVersion
 	}
 
-	logp.Debug("docker", "Negotiating client version")
-	ping, err := c.Ping(context.Background())
+	c, err := client.NewClient(host, version, httpClient, nil)
 	if err != nil {
-		logp.Debug("docker", "Failed to perform ping: %s", err)
+		return c, err
 	}
 
-	// try the latest version before versioning headers existed
-	if ping.APIVersion == "" {
-		ping.APIVersion = "1.24"
-	}
+	if os.Getenv("DOCKER_API_VERSION") == "" {
```

Comment:
Is `DOCKER_API_VERSION` set by default? If not it seems this code is always called also for a  newer version?

Reply:
DOCKER_API_VERSION is used to hardcode client API version. If it's unset (which is common), beats should negotiate client API version to ensure we use one compatible with the server